XenAdmin/Actions/GUIActions/DownloadUpdatesXmlAction.cs

```diff
@@ -313,5 +322,26 @@ protected virtual XmlDocument FetchCheckForUpdatesXml(string location)
 
             return xdoc;
         }
+
+        private string GetUniqueIdHash()
+        {
+            string uniqueIdHash = "nil";
+
+            var managementObj = new System.Management.ManagementObject("Win32_OperatingSystem=@");
+            string serialNumber = (string)managementObj["SerialNumber"];
+
+            if (!string.IsNullOrWhiteSpace(serialNumber))
+            {
+                var serialBytes = Encoding.ASCII.GetBytes("JustSomeSaltForCheckForUpdates" + serialNumber);

```

Comment:
I'm not sure what the salt is doing here. I don't see any advantage of a publicly known salt, and it just makes the md5sum longer to compute.